woocommerce/woocommerce-ios
# [IPPInAppFeedback] Add In-Person Payments feedback banner track events

Closes: #8672 
Part of: #8530 

## Description
This PR adds track events for the In-Person Payments feedback banner project.

Note: Closing the Survey without sending feedback dismisses the banner without tracking any event, this is expected as we're only tracking when a user taps on "share feedback", or the "dismiss" buttons.

## Changes
- Added new `WooAnalyticsEvent.InPersonPaymentsFeedbackBanner` that contains keys for  `campaign`, `source`, and `remind_later`. As well as static methods for when a banner is shown, tapped, and dismissed.
- Added 3 new campaigns: `ipp_not_user`, `ipp_new_user`, `ipp_power_user` which will be used on different conditions.
- Added track events for banner shown, tapped, and dismissed (temporarily, or permanently)

## Testing instructions
On a store with COD enabled (with or without IPP transactions):
1. Comment out the lines [866](https://github.com/woocommerce/woocommerce-ios/blob/6d5ff1509685b6b2b5e54af62afbc4bfe21be0f1/WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift#L866) and [877](https://github.com/woocommerce/woocommerce-ios/blob/6d5ff1509685b6b2b5e54af62afbc4bfe21be0f1/WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift#L877), so we're sure to not update the feedback status and makes testing easier.
3. Go to Orders > See the In-Person Payments feedback banner at the top of the screen
4. Confirm the `survey_banner_shown` event is tracked:
```
2023-01-20 16:25:40.933200+0800 WooCommerce[93597:3292192] 🔵 Tracked survey_banner_shown, properties: [AnyHashable("blog_id"): 121710041, AnyHashable("is_wpcom_store"): false, AnyHashable("campaign"): "ipp_not_user", AnyHashable("source"): "order_list"]
```
5. Tap on "Share Feedback". Confirm the `survey_banner_cta_tapped` event is tracked:
```
2023-01-20 16:19:11.911450+0800 WooCommerce[93181:3285720] 🔵 Tracked survey_banner_cta_tapped, properties: [AnyHashable("blog_id"): 121710041, AnyHashable("campaign"): "ipp_not_user", AnyHashable("source"): "order_list", AnyHashable("is_wpcom_store"): false]
```
6. Tap on "Dismiss" > Tap on "Remind me later". Confirm the `survey_banner_dismissed` event for `AnyHashable("remind_later"): true` is tracked:
```
2023-01-20 16:12:48.669754+0800 WooCommerce[92891:3280442] 🔵 Tracked survey_banner_dismissed, properties: [AnyHashable("campaign"): "ipp_not_user", AnyHashable("is_wpcom_store"): false, AnyHashable("remind_later"): true, AnyHashable("blog_id"): 121710041, AnyHashable("source"): "order_list"]
```
7. Tap on "Dismiss" > Tap on "Don't show again". Confirm the `survey_banner_dismissed` event for `AnyHashable("remind_later"): false` is tracked:
```
2023-01-20 16:13:19.287580+0800 WooCommerce[92891:3280712] 🔵 Tracked survey_banner_dismissed, properties: [AnyHashable("remind_later"): false, AnyHashable("source"): "order_list", AnyHashable("campaign"): "ipp_not_user", AnyHashable("blog_id"): 121710041, AnyHashable("is_wpcom_store"): false]
```

## Registered track events:
All events have been registered and validated as well:
- [x] https://github.com/Automattic/tracks-events-registration/pull/1343
- [x] https://github.com/Automattic/tracks-events-registration/pull/1344
- [x] https://github.com/Automattic/tracks-events-registration/pull/1345
<img width="506" alt="Screenshot 2023-01-23 at 16 49 10" src="https://user-images.githubusercontent.com/3812076/213998541-f59e9584-4619-460b-a6ae-d92e5daa8ff7.png">
